packages/financial-templates-lib/src/clients/InsuredBridgeL2Client.ts

```diff
@@ -122,12 +125,60 @@ export class InsuredBridgeL2Client {
     });
   }
 
-  async getFundsDepositedEvents(blockSearchConfig: { fromBlock: number; toBlock: number }): Promise<EventData[]> {
-    return await this.bridgeDepositBox.getPastEvents("FundsDeposited", blockSearchConfig);
+  async getFundsDepositedEvents(eventSearchOptions: EventSearchOptions): Promise<EventData[]> {
+    const eventsData = await getEventsForMultipleProviders(
+      [this.l2Web3].concat(this.fallbackL2Web3s),
+      this.bridgeDepositBox.options.jsonInterface,
+      this.bridgeDepositAddress,
+      "FundsDeposited",
+      eventSearchOptions
+    );
+    if (eventsData.missingEvents.length > 0) {
+      const errorMessage = `L2 RPC endpoints disagree about L2 contract events, please manually investigate.`;
+      const error = new Error(errorMessage);
+      this.logger.error({
+        at: "InsuredBridgeL2Client",
+        message:
+          "L2 client will not complete update because L2 RPC endpoints disagree about L2 contract events. This error requires manual investigation to determine which events are valid and whether manual disputes should be sent. L2 rpcs are described in NODE_URL and RETRY_CONFIG environment variables.",
+        eventName: "FundsDeposited",
+        eventSearchOptions,
+        countMissingEvents: eventsData.missingEvents.length,
+        countMatchingEvents: eventsData.events.length,
+        error,
+      });
+      throw error;
+    }
+
+    // All events were found in all providers, can return any of the event data arrays
+    return eventsData.events;
   }
 
-  async getWhitelistTokenEvents(blockSearchConfig: { fromBlock: number; toBlock: number }): Promise<EventData[]> {
-    return await this.bridgeDepositBox.getPastEvents("WhitelistToken", blockSearchConfig);
+  async getWhitelistTokenEvents(eventSearchOptions: EventSearchOptions): Promise<EventData[]> {
+    const eventsData = await getEventsForMultipleProviders(
+      [this.l2Web3].concat(this.fallbackL2Web3s),
+      this.bridgeDepositBox.options.jsonInterface,
+      this.bridgeDepositAddress,
+      "WhitelistToken",
+      eventSearchOptions
+    );
+    if (eventsData.missingEvents.length > 0) {
+      const errorMessage = `L2 RPC endpoints disagree about L2 contract events, please manually investigate.`;
+      const error = new Error(errorMessage);
+      this.logger.error({
+        at: "InsuredBridgeL2Client",
+        message:
```

Comment:
this is a long message! it might be good to split this up. perhaps make the `message` key short. something like: `L2 RPC endpoint state disagreement! 🤺` then put a different key for the "solution" or "moreDetails". the reason I say you should do it like this is two fold:
1) you need an emoji on all slack logs ;)
2) the "message" field is sent in the top line of the slack message. the should be short and direct. This is analogous to the `Relay instantly sent` for the relay. the "details" should follow below the message

Reply:
ugh you're totally right i forgot about emojis